### PR TITLE
[CI] Modify Jenkinfile to always display junit report, fix for #8674

### PR DESCRIPTION
**Pull request by mikepapadim**
This is a fix for #8674.

It always displays `junit` reports when tests fail. 

@areusch

### CODE REVIEW

**Comment by areusch** [comment]
thanks @mikepapadim , pushed to https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/ci-docker-staging/144/pipeline with some injected test failures to see how they look!

**Comment by areusch** [comment]
hmm https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/ci-docker-staging/144/tests

don't see any tests

**Comment by mikepapadim** [comment]
I will check again. An alternative seems to be moving the junit as the following post-stage action:
```
 post {
        always {
            junit "build/pytest-results/*.xml"
        }
    }
```

**Comment by areusch** [comment]
@mikepapadim I think this might still be busted. can you see if you can run it through a staging jenkins? [here's](https://github.com/areusch/incubator-tvm/tree/jenkinsfile_upd_test_failure) what i am using to test your change

**Comment by areusch** [comment]
@mikepapadim any update here?

**Comment by mikepapadim** [comment]
> @mikepapadim any update here?

Need to give it a go through CI. I ll update accordingly.

**Comment by mikepapadim** [comment]
Let's wait for the CI, but now with the post-action per stage, test results should always be registerd.

File: README.md [hunk 1]
```diff
@@ -26,8 +26,11 @@
 [![WinMacBuild](https://github.com/apache/tvm/workflows/WinMacBuild/badge.svg)](https://github.com/apache/tvm/actions?query=workflow%3AWinMacBuild)
 
 Apache TVM is a compiler stack for deep learning systems. It is designed to close the gap between the
-productivity-focused deep learning frameworks, and the performance- and efficiency-focused hardware backends.
-TVM works with deep learning frameworks to provide end to end compilation to different backends.
+ASDADA
```
Comment (areusch) [inline README.md:29]: there are a couple of random changes to `.md` files in here, can you revert?

**Review by areusch — CHANGES_REQUESTED** [review]
thanks @mikepapadim i think the Jenkinsfile looks good now, just some extra cruft in this one. I pushed to https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/ci-docker-staging/159/pipeline with some injected test failures.

can you remove the cruft and assuming that build looks good (test failures are reported) we can merge?

**Comment by mikepapadim** [comment]
> thanks @mikepapadim i think the Jenkinsfile looks good now, just some extra cruft in this one. I pushed to https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/ci-docker-staging/159/pipeline with some injected test failures.
> 
> can you remove the cruft and assuming that build looks good (test failures are reported) we can merge?

I will fix them. Thanks for reviewing. 

Btw if we go forward with #9042 then we might want to wait for this one as it could potentially be much cleaner with a declarive pipeline.

**Comment by areusch** [comment]
@mikepapadim let's merge this first, then migrate to #9042 separately. it will be more helpful to get test flakiness data sooner rather than later.

**Comment by denise-k** [comment]
@areusch @mikepapadim any update on this?

**Comment by driazati** [comment]
With something like #9823 this could also help this be cleaner (since the `post` could be defined just once)

**Comment by areusch** [comment]
closing due to stale PR and I believe this is implemented now

**Pull request closed without merging**